packages/ragbits-document-search/src/ragbits/document_search/retrieval/rephrasers/llm.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional
+
+from ragbits.core.llms import get_llm
+from ragbits.core.llms.base import LLM, LLMOptions
+from ragbits.core.prompt import Prompt
+from ragbits.document_search.retrieval.rephrasers.base import QueryRephraser
+from ragbits.document_search.retrieval.rephrasers.prompt_query_rephraser import QueryRephraserPrompt, _PromptInput
+
+
+class LLMQueryRephraser(QueryRephraser):
+    """A rephraser class that uses a LLM to rephrase queries."""
+
+    def __init__(self, llm: LLM):
+        """
+        Initialize the LLMQueryRephraser with a LLM.
+
+        Args:
+            llm: A LLM instance to handle query rephrasing.
+        """
+
+        self._llm = llm
+
+    async def rephrase(
+        self, query: Optional[str] = None, prompt: Optional[Prompt] = None, options: Optional[LLMOptions] = None
+    ) -> list[str]:
+        """
+        Rephrase a given query using the LLM.
+
+        Args:
+            query: The query to be rephrased. If not provided, a custom prompt must be given.
+            prompt: A prompt object defining how the rephrasing should be done.
+                    If not provided, the default `QueryRephraserPrompt` is used, along with the provided query.
+            options: Optional settings for the LLM to control generation behavior.
+
+        Returns:
+            A list containing the rephrased query.
+
+        Raises:
+            ValueError: If both `query` and `prompt` are None.
+        """
+
+        if query is None and prompt is None:
```

Comment:
I don't understand this check. The `query` should always be present because it's the fundamental input for the rephrasing process. As for `prompt`, I'd look at it as a rephrasing strategy, you can have a prompt that generates sub-queries, step-back or even hyde (they all work on the query as input).

Reply:
fixed